src/test/css/nodes.test.ts

```diff
@@ -24,23 +24,35 @@ export function assertNodes(fn: (input: string) => nodes.Node, input: string, ex
 
 	node.acceptVisitor(visitor);
 
-	let actual = visitor.tree.join(',') + ',';
+	let actual = visitor.tree;
+	let actualStr = actual.join(',');
 	let segments = expected.split(',');
-	let oldIndex: number | undefined = undefined;
-	let index = -1;
 
 	while (segments.length > 0) {
-		let segment = segments.shift()!;
-		if (segment === '...') {
+		let expectedSegment = segments.shift()!;
+		let actualSegment = actual.shift()!;
+
+		if (expectedSegment === '...') {
+			let nextExpectedSegment = segments[0];
+			let nextActualSegment = actual[0];
+
+			while (actual.length > 0) {
+				if (nextExpectedSegment === nextActualSegment) {
+					break;
+				}
+
+				actualSegment = actual.shift()!;
+				nextActualSegment = actual[0];
+			}
+
```

Comment:
This test helper was broken, it didn't actually test anything.
Adding any `...` in a expect string would cause the previous algorithm to just skip to the end in case of issues.

This new algorithm is a bit stricter, making it easier to write reliable tests for nested and repeating structures.